### PR TITLE
Add a timestamp to publication acks

### PR DESCRIPTION
**Pull request by Jmgr**
Publication ACKs currently return the stream offset the message was committed to, but not its timestamp. This PR adds this.

### CODE REVIEW

File: server/partition.go [hunk 1]
```diff
@@ -732,15 +732,16 @@ func (p *partition) messageProcessingLoop(recvChan <-chan *nats.Msg, stop <-chan
...
-		CorrelationId:    msg.CorrelationID,
-		AckPolicy:        msg.AckPolicy,
+		Stream:             p.Stream,
+		PartitionSubject:   p.Subject,
+		MsgSubject:         string(msg.Headers["subject"]),
+		Offset:             offset,
+		AckInbox:           msg.AckInbox,
+		CorrelationId:      msg.CorrelationID,
+		AckPolicy:          msg.AckPolicy,
+		ReceptionTimestamp: timestamp,
```
Comment (tylertreat) [inline server/partition.go:744]: Is there a reason you're passing back the timestamps from `commitlog.Append` and then passing them into `processPendingMessage`? I think you could leave the commitlog API unchanged and just do the following here:

```go
ReceptionTimestamp: msg.Timestamp,
```

The `commitlog.Message` has the [receive timestamp on it](https://github.com/liftbridge-io/liftbridge/blob/master/server/commitlog/message.go#L22).

**Review by tylertreat — COMMENTED** [review]
You can also update liftbridge-api now that it's merged.

File: server/partition.go (hunk 1 above)
Comment (Jmgr) [reply server/partition.go]: Good point. I've updated the PR.

**Comment by tylertreat** [comment]
For some reason this is causing a test to fail. I'm looking into it...

**Pull request merged**